src/datasets/containers.jl

```diff
@@ -46,4 +48,36 @@ isimagefile(file::String) = occursin(IMAGEFILE_REGEX, lowercase(file))
 const IMAGEFILE_REGEX = r"\.(gif|jpe?g|tiff?|png|webp|bmp)$"
 
 
-## TODO: TableDataset
+struct TableDataset{T}
+    table::T #Should implement Tables.jl interface
+    TableDataset{T}(table::T) where T = Tables.istable(table) ? new{T}(table) : error("Object doesn't implement Tables.jl interface")
+end
+
+TableDataset(path::String) = TableDataset(DataFrame(CSV.File(path)))
```

Comment:
Reflecting Ari's comment from Zulip, maybe we can just use `CSV.File` here.

Reply:
Okay I can change it so that the default behaviour on getting a path is to use `CSV.File` object. It's just that I thought having a familiar API like DataFrames might be easier to work with, specially for people coming from Python fastai.

Reply:
That's true, but I think in the this case it is more important to be performant reading from disk. Most people won't be directly interacting with the underlying table.